### PR TITLE
Fixes for value analysis

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Fill `op->reg` for ARM64 `br`/`blr` analysis
* Fill `op->reg` for ARM `bx`/`blx` analysis
* Fill `RzAnalysisValue->type`
* Use only `RzAnalysisValue` in `rz_core_link_stroff()`

**Test plan**

Tests pass

**Closing issues**

None

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]
Changes look ok. Do we have at least a simple test case for these changes as well?

**Review by wargio — APPROVED** [review]

**Comment by XVilka** [comment]
@GustavoLCR could you please add a test or two specifically for these changes if possible?

**Comment by XVilka** [comment]
```
../test/unit/test_analysis_op.c: In function ‘test_rz_analysis_op_val’:
../test/unit/test_analysis_op.c:16:45: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   16 |  int len = rz_analysis_op(analysis, &op, 0, "\x48\x8b\x44\x0b\x04", 5, RZ_ANALYSIS_OP_MASK_VAL);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~
      |                                             |
      |                                             char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:25:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   25 |  len = rz_analysis_op(analysis, &op, 0, "\x48\xc7\xc0\x04\x00\x00\x00", 7, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:34:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   34 |  len = rz_analysis_op(analysis, &op, 0, "\x41\x68\x63\xf8", 4, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:43:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   43 |  len = rz_analysis_op(analysis, &op, 0, "\x01\x32\x80\xd2", 4, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:53:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   53 |  len = rz_analysis_op(analysis, &op, 0, "\x03\x10\x92\xe7", 4, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:64:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   64 |  len = rz_analysis_op(analysis, &op, 0, "\xd1\x58", 2, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:75:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   75 |  len = rz_analysis_op(analysis, &op, 0, "\x06\x4d", 2, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:84:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   84 |  len = rz_analysis_op(analysis, &op, 0, "\x22\xc5", 2, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
cc1: all warnings being treated as errors
```

**Review by ret2libc — APPROVED** [review]

File: test/unit/test_analysis_op.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#define SWITCH_TO_ARCH_BITS(arch, bits) \
+	rz_analysis_use(analysis, arch); \
+	rz_analysis_set_bits(analysis, bits);
+
+bool test_rz_analysis_op_val() {
+	RzAnalysis *analysis = rz_analysis_new();
+	RzAnalysisOp op;
+	SWITCH_TO_ARCH_BITS("x86", 64);
+	// mov rax, [rbx+rcx+4]
+	int len = rz_analysis_op(analysis, &op, 0, "\x48\x8b\x44\x0b\x04", 5, RZ_ANALYSIS_OP_MASK_VAL);
```
Comment (wargio) [inline test/unit/test_analysis_op.c]:
```suggestion
	int len = rz_analysis_op(analysis, &op, 0, (const ut8 *)"\x48\x8b\x44\x0b\x04", 5, RZ_ANALYSIS_OP_MASK_VAL);
```

**Review by wargio — CHANGES_REQUESTED** [review]
add `(const ut8 *)` in `test_analysis_op.c` so we can fix the warning

**Comment by XVilka** [comment]
```
../test/unit/test_analysis_op.c: In function ‘test_rz_analysis_op_val’:
../test/unit/test_analysis_op.c:25:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   25 |  len = rz_analysis_op(analysis, &op, 0, "\x48\xc7\xc0\x04\x00\x00\x00", 7, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:34:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   34 |  len = rz_analysis_op(analysis, &op, 0, "\x41\x68\x63\xf8", 4, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:43:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   43 |  len = rz_analysis_op(analysis, &op, 0, "\x01\x32\x80\xd2", 4, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:53:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   53 |  len = rz_analysis_op(analysis, &op, 0, "\x03\x10\x92\xe7", 4, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:64:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   64 |  len = rz_analysis_op(analysis, &op, 0, "\xd1\x58", 2, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:75:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   75 |  len = rz_analysis_op(analysis, &op, 0, "\x06\x4d", 2, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
../test/unit/test_analysis_op.c:84:41: error: pointer targets in passing argument 4 of ‘rz_analysis_op’ differ in signedness [-Werror=pointer-sign]
   84 |  len = rz_analysis_op(analysis, &op, 0, "\x22\xc5", 2, RZ_ANALYSIS_OP_MASK_VAL);
      |                                         ^~~~~~~~~~
      |                                         |
      |                                         char *
In file included from ../test/unit/test_analysis_op.c:4:
../librz/include/rz_analysis.h:1415:89: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
 1415 | RZ_API int rz_analysis_op(RzAnalysis *analysis, RzAnalysisOp *op, ut64 addr, const ut8 *data, int len, RzAnalysisOpMask mask);
      |                                                                              ~~~~~~~~~~~^~~~
cc1: all warnings being treated as errors
```

**Review by wargio — APPROVED** [review]

**Pull request merged**